dashboard/build.bash

```
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Builds dashboard project
+#
+# Optional Flags:
+# -t        Run tests only
+# -d [env]  Build and deploy to specified Azure Environment (e.g. tts/dev)
+#
+# Usage:
+# ./build.bash
+# ./build.bash -t
+# ./build.bash -d tts/dev
+
+source $(dirname "$0")/../tools/common.bash || exit
+
+# set modes
+test_mode="false"
+azure_env="none"
+
+while getopts ":td:" arg; do
+	case "${arg}" in
+		t) test_mode="true" ;;
+    d) azure_env=$OPTARG ;;
+	esac
+done
+# end set modes
+
+set_constants () {
+  DASHBOARD_APP_NAME=$PREFIX-app-dashboard-$ENV # TODO: make this DRY
+}
+
+run_tests () {
+  echo "Running tests"
+  dotnet test
+}
+
+run_deploy () {
+  azure_env=$1
+  source $(dirname "$0")/../iac/env/${azure_env}.bash
+  source $(dirname "$0")/../iac/iac-common.bash
+  verify_cloud
```

You want an CLI option error to fail fast, ideally before any amount of work (e.g., a build) is done. This comment may be moot if the `dotnet build` gets removed on deploys.